src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4167,45 +3952,29 @@ char* os::pd_reserve_memory_special(size_t bytes, size_t alignment, size_t page_
   return addr;
 }
 
-bool os::Linux::release_memory_special_shm(char* base, size_t bytes) {
-  // detaching the SHM segment will also delete it, see reserve_memory_special_shm()
-  return shmdt(base) == 0;
-}
-
-bool os::Linux::release_memory_special_huge_tlbfs(char* base, size_t bytes) {
-  return pd_release_memory(base, bytes);
-}
-
 bool os::pd_release_memory_special(char* base, size_t bytes) {
   assert(UseLargePages, "only for large pages");
-  bool res;
-
-  if (UseSHM) {
-    res = os::Linux::release_memory_special_shm(base, bytes);
-  } else {
-    assert(UseHugeTLBFS, "must be");
-    res = os::Linux::release_memory_special_huge_tlbfs(base, bytes);
-  }
-  return res;
+  // Plain munmap is sufficient
+  return pd_release_memory(base, bytes);
 }
 
 size_t os::large_page_size() {
   return _large_page_size;
 }
 
-// With SysV SHM the entire memory region must be allocated as shared
-// memory.
-// HugeTLBFS allows application to commit large page memory on demand.
-// However, when committing memory with HugeTLBFS fails, the region
+// static hugepages (hugetlbfs) allow application to commit large page memory
+// on demand.
+// However, when committing memory with hugepages fails, the region
 // that was supposed to be committed will lose the old reservation
 // and allow other threads to steal that memory region. Because of this
-// behavior we can't commit HugeTLBFS memory.
+// behavior we can't commit hugetlbfs memory. Instead, we commit that
+// memory at reservation.
 bool os::can_commit_large_page_memory() {
   return UseTransparentHugePages;
 }
 
 bool os::can_execute_large_page_memory() {
-  return UseTransparentHugePages || UseHugeTLBFS;
+  return UseTransparentHugePages;
```

Comment:
This change introduced a regression: https://bugs.openjdk.org/browse/JDK-8319795
PR fixing the regression: https://github.com/openjdk/jdk/pull/16582